week-3/todo-list/script.js

```diff
@@ -1,25 +1,33 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
-
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
-
-populateTodoList(todos);
+let list = document.getElementById("todo-list");
+const addTodo = document.querySelector("#add");
+const todos = [];
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
+function addNewTodo() {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const input = document.querySelector('input');
+  if (input.value != "") {
```

Comment:
We generally try to avoid `==` and `!=`, instead using `===` or `!==` - can you talk through the difference, and why we prefer the longer ones?

Reply:
Yes. === and !== check both values and their types. The reason is to avoid unexpected type coercion. Using strict equality makes the code more predictable and less error-prone. When you explicitly compare both value and type, you reduce the risk of unintended consequences that can arise from automatic type conversions.